lib/routes/utils/request_helper.js

```diff
@@ -9,9 +9,52 @@
  * @returns {boolean}
  */
 function wantsKeys (request) {
-  return request.query.keys === 'true'
+  return request.query.keys === 'true' || request.query.keys === true
+}
+
+/**
+ * Returns whether or not to perform a signin verification email.
+ *
+ * @param account
+ * @param config
+ * @param request
+ * @returns {boolean}
+ */
+function shouldEnableSigninConfirmation(account, config, request) {
+
+  var confirmLogin = config.signinConfirmation && config.signinConfirmation.enabled
+  if (confirmLogin) {
+
+    var uid = account.uid.toString('hex')
+    var email = account.email
+
+    // If feature enabled, always enable for these emails
+    var emailDomain = account.email.substring(email.indexOf('@'), email.length).toLocaleLowerCase()
+    var isValidEmail = config.signinConfirmation.allowEmails.indexOf(emailDomain) > -1
+    if (isValidEmail) {
+      return true
+    }
```

Comment:
More broadly, this function might be easier to follow as a single level of short-circuiting returns, rather than nested if-elses.  For example:

```
if (it's not enabled in the config) {
  return false
}

if (the context value is not in the list of supported clients) {
  return false
}

if (the email is on the list of forced emails) {
  return true
}

calculate the sample rate
return sample rate < the configured rate
```